### PR TITLE
Refactor simple Unit Tests

### PR DESCRIPTION
**Pull request by Davfon**
I have refactored some simple unit tests to make them easier to understand, more readable, more maintainable and less likely for human error. Some identified [test smells](https://testsmells.org/pages/testsmells.html#GeneralFixture) (like General Fixture and Test Code Duplication) have been fixed. 
It addresses #6207, although it is mainly a refactoring. 

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java [hunk 1]
```diff
@@ -3,24 +3,13 @@
...
-    @BeforeEach
-    public void setUp() {
-        comparator = new CrossRefEntryComparator();
-    }
-
-    @AfterEach
-    public void tearDown() {
-        comparator = null;
-    }
+    private static final CrossRefEntryComparator comparator = new CrossRefEntryComparator();
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: Why did you change this? I think it's good practice to initialize the object under test in a setup method, so that e.g. changes to it in one test doesn't leak to other tests.

**Comment by Siedlerchr** [comment]
It's nice that you are interested in contributing to JabRef, however, it would be nice if you not only focused on simple technical issues, but on issues that affect the users.

File: src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java (hunk 1 above)
Comment (Davfon) [reply src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: If I see it correctly it is not possible to change the CrossRefEntryComparator Object after creation. In other words there are no setter methods and the constructor does not take any arguments. 

That is why I think the beforeeach is superfluous. Why create a new object, if the one created can be used?

On the other hand, I get your point. If changes to this class are done, like adding arguments to the constructor or adding setter methods, then the BeforeEach makes sense.

Comment (calixtus) [reply src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: Creation of objects I java is extremely efficient. This shouldn't be a concern.

Comment (koppor) [reply src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: In tests, one just does not use `static`.

- Possiblity one: class variable (you can go for that, because the `setup` just does that) --> move the initialization code up.
- Possibility two: use `setup` (as in the code)

File: src/test/java/org/jabref/logic/bst/BibtexCaseChangersTest.java [hunk 2]
```diff
@@ -1,134 +1,156 @@
...
+                Arguments.of("this is a simple example {TITLE}", "This is a simple example {TITLE}"),
+                Arguments.of("this {IS} another simple example tit{LE}", "This {IS} another simple example tit{LE}"),
+                Arguments.of("{What ABOUT thIS} one?", "{What ABOUT thIS} one?"),
+                Arguments.of("{And {thIS} might {a{lso}} be possible}", "{And {thIS} might {a{lso}} be possible}")
+        );
     }
 
-    private void assertCaseChangerAllLowers(final String string, final String string2) {
-        assertEquals(string, BibtexCaseChanger.changeCase(string2, FORMAT_MODE.ALL_LOWERS));
+    /* the real test would look like as follows. Also from the comment of issue 176, order reversed as the "should be" comes first
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bst/BibtexCaseChangersTest.java:140]: Please add `@Disabled` and do not work with commented code.

File: src/test/java/org/jabref/logic/bst/BibtexPurifyTest.java [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+                Arguments.of("oe", "{\\oe}"),
+                Arguments.of("Hi oeHi ", "Hi {\\oe   }Hi "),
+                Arguments.of("Jonathan Meyer and Charles Louis Xavier Joseph de la Vallee Poussin", "Jonathan Meyer and Charles Louis Xavier Joseph de la Vall{\\'e}e Poussin"),
+                Arguments.of("e", "{\\'e}"),
+                Arguments.of("Edouard Masterly", "{\\'{E}}douard Masterly"),
+                Arguments.of("Ulrich Underwood and Ned Net and Paul Pot", "Ulrich {\\\"{U}}nderwood and Ned {\\~N}et and Paul {\\={P}}ot")
+        );
     }
 
     private void assertPurify(final String string, final String string2) {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bst/BibtexPurifyTest.java:34]: Please integrate in `testPurify`. We don't see why this private (!) method is there.

File: src/test/java/org/jabref/logic/formatter/bibtexfields/AddBracesFormatterTest.java [hunk 4]
```diff
@@ -10,12 +9,7 @@
  */
 class AddBracesFormatterTest {
 
-    private AddBracesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new AddBracesFormatter();
-    }
+    private static final AddBracesFormatter formatter = new AddBracesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/AddBracesFormatterTest.java:12]: nononono static

```suggestion
    private AddBracesFormatter formatter = new AddBracesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatterTest.java [hunk 5]
```diff
@@ -10,12 +9,7 @@
  */
 class CleanupUrlFormatterTest {
 
-    private CleanupUrlFormatter formatter;
-
-    @BeforeEach
-    void setUp() {
-        formatter = new CleanupUrlFormatter();
-    }
+    private static final CleanupUrlFormatter formatter = new CleanupUrlFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatterTest.java:12]:
```suggestion
    private CleanupUrlFormatter formatter = new CleanupUrlFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/ClearFormatterTest.java [hunk 6]
```diff
@@ -10,12 +9,7 @@
  */
 public class ClearFormatterTest {
 
-    private ClearFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new ClearFormatter();
-    }
+    private static final ClearFormatter formatter = new ClearFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/ClearFormatterTest.java:12]:
```suggestion
    private ClearFormatter formatter = new ClearFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeAmpersandsFormatterTest.java [hunk 7]
```diff
@@ -1,18 +1,12 @@
...
 
 class EscapeAmpersandsFormatterTest {
 
-    private EscapeAmpersandsFormatter formatter;
-
-    @BeforeEach
-    void setUp() {
-        formatter = new EscapeAmpersandsFormatter();
-    }
+    private static final EscapeAmpersandsFormatter formatter = new EscapeAmpersandsFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeAmpersandsFormatterTest.java:9]:
```suggestion
    private EscapeAmpersandsFormatter formatter = new EscapeAmpersandsFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeUnderscoresFormatterTest.java [hunk 8]
```diff
@@ -1,18 +1,12 @@
...
 
 class EscapeUnderscoresFormatterTest {
 
-    private EscapeUnderscoresFormatter formatter;
-
-    @BeforeEach
-    void setUp() {
-        formatter = new EscapeUnderscoresFormatter();
-    }
+    private static final EscapeUnderscoresFormatter formatter = new EscapeUnderscoresFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/EscapeUnderscoresFormatterTest.java:9]:
```suggestion
    private EscapeUnderscoresFormatter formatter = new EscapeUnderscoresFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatterTest.java [hunk 9]
```diff
@@ -10,12 +9,7 @@
  */
 public class HtmlToLatexFormatterTest {
 
-    private HtmlToLatexFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new HtmlToLatexFormatter();
-    }
+    private static final HtmlToLatexFormatter formatter = new HtmlToLatexFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatterTest.java:12]:
```suggestion
    private HtmlToLatexFormatter formatter = new HtmlToLatexFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/HtmlToUnicodeFormatterTest.java [hunk 10]
```diff
@@ -1,18 +1,12 @@
...
 
 public class HtmlToUnicodeFormatterTest {
 
-    private HtmlToUnicodeFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new HtmlToUnicodeFormatter();
-    }
+    private static final HtmlToUnicodeFormatter formatter = new HtmlToUnicodeFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/HtmlToUnicodeFormatterTest.java:9]:
```suggestion
    private HtmlToUnicodeFormatter formatter = new HtmlToUnicodeFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/LatexCleanupFormatterTest.java [hunk 11]
```diff
@@ -1,18 +1,12 @@
...
 
 class LatexCleanupFormatterTest {
 
-    private LatexCleanupFormatter formatter;
-
-    @BeforeEach
-    void setUp() {
-        formatter = new LatexCleanupFormatter();
-    }
+    private static final LatexCleanupFormatter formatter = new LatexCleanupFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/LatexCleanupFormatterTest.java:9]:
```suggestion
    private LatexCleanupFormatter formatter = new LatexCleanupFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeDateFormatterTest.java [hunk 12]
```diff
@@ -10,12 +9,7 @@
  */
 public class NormalizeDateFormatterTest {
 
-    private NormalizeDateFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new NormalizeDateFormatter();
-    }
+    private static final NormalizeDateFormatter formatter = new NormalizeDateFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeDateFormatterTest.java:12]:
```suggestion
    private NormalizeDateFormatter formatter = new NormalizeDateFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeEnDashesFormatterTest.java [hunk 13]
```diff
@@ -10,12 +9,7 @@
  */
 public class NormalizeEnDashesFormatterTest {
 
-    private NormalizeEnDashesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new NormalizeEnDashesFormatter();
-    }
+    private static final NormalizeEnDashesFormatter formatter = new NormalizeEnDashesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeEnDashesFormatterTest.java:12]:
```suggestion
    private NormalizeEnDashesFormatter formatter = new NormalizeEnDashesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeMonthFormatterTest.java [hunk 14]
```diff
@@ -10,12 +9,7 @@
  */
 public class NormalizeMonthFormatterTest {
 
-    private NormalizeMonthFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new NormalizeMonthFormatter();
-    }
+    private static final NormalizeMonthFormatter formatter = new NormalizeMonthFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeMonthFormatterTest.java:12]:
```suggestion
    private NormalizeMonthFormatter formatter = new NormalizeMonthFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeNamesFormatterTest.java [hunk 15]
```diff
@@ -10,12 +9,7 @@
  */
 public class NormalizeNamesFormatterTest {
 
-    private NormalizeNamesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new NormalizeNamesFormatter();
-    }
+    private static final NormalizeNamesFormatter formatter = new NormalizeNamesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizeNamesFormatterTest.java:12]:
```suggestion
    private NormalizeNamesFormatter formatter = new NormalizeNamesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizePagesFormatterTest.java [hunk 16]
```diff
@@ -13,7 +13,7 @@
  */
 public class NormalizePagesFormatterTest {
 
-    private final NormalizePagesFormatter formatter = new NormalizePagesFormatter();
+    private static final NormalizePagesFormatter formatter = new NormalizePagesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/NormalizePagesFormatterTest.java:16]:
```suggestion
    private final NormalizePagesFormatter formatter = new NormalizePagesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/OrdinalsToSuperscriptFormatterTest.java [hunk 17]
```diff
@@ -10,12 +9,7 @@
  */
 public class OrdinalsToSuperscriptFormatterTest {
 
-    private OrdinalsToSuperscriptFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new OrdinalsToSuperscriptFormatter();
-    }
+    private static final OrdinalsToSuperscriptFormatter formatter = new OrdinalsToSuperscriptFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/OrdinalsToSuperscriptFormatterTest.java:12]:
```suggestion
    private OrdinalsToSuperscriptFormatter formatter = new OrdinalsToSuperscriptFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveBracesFormatterTest.java [hunk 18]
```diff
@@ -10,12 +9,7 @@
  */
 public class RemoveBracesFormatterTest {
 
-    private RemoveBracesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new RemoveBracesFormatter();
-    }
+    private static final RemoveBracesFormatter formatter = new RemoveBracesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveBracesFormatterTest.java:12]:
```suggestion
    private RemoveBracesFormatter formatter = new RemoveBracesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveDigitsFormatterTest.java [hunk 19]
```diff
@@ -1,18 +1,12 @@
...
 
 public class RemoveDigitsFormatterTest {
 
-    private RemoveDigitsFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new RemoveDigitsFormatter();
-    }
+    private static final RemoveDigitsFormatter formatter = new RemoveDigitsFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveDigitsFormatterTest.java:9]:
```suggestion
    private RemoveDigitsFormatter formatter = new RemoveDigitsFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveNewlinesFormatterTest.java [hunk 20]
```diff
@@ -1,19 +1,13 @@
...
 
 public class RemoveNewlinesFormatterTest {
 
-    private RemoveNewlinesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new RemoveNewlinesFormatter();
-    }
+    private static final RemoveNewlinesFormatter formatter = new RemoveNewlinesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveNewlinesFormatterTest.java:10]:
```suggestion
    private RemoveNewlinesFormatter formatter = new RemoveNewlinesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveRedundantSpacesFormatterTest.java [hunk 21]
```diff
@@ -1,18 +1,12 @@
...
 
 public class RemoveRedundantSpacesFormatterTest {
 
-    private RemoveRedundantSpacesFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new RemoveRedundantSpacesFormatter();
-    }
+    private static final RemoveRedundantSpacesFormatter formatter = new RemoveRedundantSpacesFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/RemoveRedundantSpacesFormatterTest.java:9]:
```suggestion
    private RemoveRedundantSpacesFormatter formatter = new RemoveRedundantSpacesFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/ReplaceTabsBySpaceFormaterTest.java [hunk 22]
```diff
@@ -1,19 +1,12 @@
...
 
 public class ReplaceTabsBySpaceFormaterTest {
 
-    private ReplaceTabsBySpaceFormater formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new ReplaceTabsBySpaceFormater();
-    }
+    private static final ReplaceTabsBySpaceFormater formatter = new ReplaceTabsBySpaceFormater();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/ReplaceTabsBySpaceFormaterTest.java:9]:
```suggestion
    private ReplaceTabsBySpaceFormater formatter = new ReplaceTabsBySpaceFormater();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/ShortenDOIFormatterTest.java [hunk 23]
```diff
@@ -2,20 +2,14 @@
...
 @FetcherTest
 class ShortenDOIFormatterTest {
 
-    private ShortenDOIFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new ShortenDOIFormatter();
-    }
+    private static final ShortenDOIFormatter formatter = new ShortenDOIFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/ShortenDOIFormatterTest.java:12]:
```suggestion
    private ShortenDOIFormatter formatter = new ShortenDOIFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/TrimWhitespaceFormatterTest.java [hunk 24]
```diff
@@ -1,19 +1,12 @@
...
 
 public class TrimWhitespaceFormatterTest {
 
-    private TrimWhitespaceFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new TrimWhitespaceFormatter();
-    }
+    private static final TrimWhitespaceFormatter formatter = new TrimWhitespaceFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/TrimWhitespaceFormatterTest.java:9]:
```suggestion
    private TrimWhitespaceFormatter formatter = new TrimWhitespaceFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/UnicodeConverterFormatterTest.java [hunk 25]
```diff
@@ -1,21 +1,15 @@
...
-public class UnicodeConverterTest {
+public class UnicodeConverterFormatterTest {
 
-    private UnicodeToLatexFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new UnicodeToLatexFormatter();
-    }
+    private static final UnicodeToLatexFormatter formatter = new UnicodeToLatexFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/UnicodeConverterFormatterTest.java:12]:
```suggestion
    private UnicodeToLatexFormatter formatter = new UnicodeToLatexFormatter();
```

File: src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java (hunk 1 above)
Comment (koppor) [reply src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: Reason for no static: Each test has to be independed of the last test. Thus, everything under test has to be in the initial state.

File: src/test/java/org/jabref/logic/formatter/minifier/MinifyNameListFormatterTest.java [hunk 26]
```diff
@@ -10,12 +9,7 @@
  */
 public class MinifyNameListFormatterTest {
 
-    private MinifyNameListFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new MinifyNameListFormatter();
-    }
+    private static final MinifyNameListFormatter formatter = new MinifyNameListFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/minifier/MinifyNameListFormatterTest.java:12]:
```suggestion
    private MinifyNameListFormatter formatter = new MinifyNameListFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/UpperCaseFormatterTest.java [hunk 27]
```diff
@@ -10,12 +9,7 @@
  */
 public class UpperCaseFormatterTest {
 
-    private UpperCaseFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new UpperCaseFormatter();
-    }
+    private static final UpperCaseFormatter formatter = new UpperCaseFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/UpperCaseFormatterTest.java:12]:
```suggestion
    private UpperCaseFormatter formatter = new UpperCaseFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/UnprotectTermsFormatterTest.java [hunk 28]
```diff
@@ -14,7 +14,7 @@
  */
 public class UnprotectTermsFormatterTest {
 
-    private UnprotectTermsFormatter formatter = new UnprotectTermsFormatter();
+    private static final UnprotectTermsFormatter formatter = new UnprotectTermsFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/UnprotectTermsFormatterTest.java:17]:
```suggestion
    private UnprotectTermsFormatter formatter = new UnprotectTermsFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/TitleCaseFormatterTest.java [hunk 29]
```diff
@@ -13,7 +13,7 @@
  */
 public class TitleCaseFormatterTest {
 
-    private final TitleCaseFormatter formatter = new TitleCaseFormatter();
+    private static final TitleCaseFormatter formatter = new TitleCaseFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/TitleCaseFormatterTest.java:16]:
```suggestion
    private TitleCaseFormatter formatter = new TitleCaseFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/SentenceCaseFormatterTest.java [hunk 30]
```diff
@@ -13,7 +13,7 @@
  */
 public class SentenceCaseFormatterTest {
 
-    private final SentenceCaseFormatter formatter = new SentenceCaseFormatter();
+    private static final SentenceCaseFormatter formatter = new SentenceCaseFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/SentenceCaseFormatterTest.java:16]:
```suggestion
    private SentenceCaseFormatter formatter = new SentenceCaseFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/ProtectTermsFormatterTest.java [hunk 31]
```diff
@@ -15,14 +14,9 @@
...
 
-    private ProtectTermsFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new ProtectTermsFormatter(
-                new ProtectedTermsLoader(new ProtectedTermsPreferences(ProtectedTermsLoader.getInternalLists(),
-                        Collections.emptyList(), Collections.emptyList(), Collections.emptyList())));
-    }
+    private static final ProtectTermsFormatter formatter = new ProtectTermsFormatter(
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/ProtectTermsFormatterTest.java:17]:
```suggestion
    private ProtectTermsFormatter formatter = new ProtectTermsFormatter(
```

File: src/test/java/org/jabref/logic/formatter/casechanger/LowerCaseFormatterTest.java [hunk 32]
```diff
@@ -10,12 +9,7 @@
  */
 public class LowerCaseFormatterTest {
 
-    private LowerCaseFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new LowerCaseFormatter();
-    }
+    private static final LowerCaseFormatter formatter = new LowerCaseFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/LowerCaseFormatterTest.java:12]:
```suggestion
    private LowerCaseFormatter formatter = new LowerCaseFormatter();
```

File: src/test/java/org/jabref/logic/formatter/casechanger/CapitalizeFormatterTest.java [hunk 33]
```diff
@@ -12,12 +11,7 @@
  */
 public class CapitalizeFormatterTest {
 
-    private CapitalizeFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new CapitalizeFormatter();
-    }
+    private static final CapitalizeFormatter formatter = new CapitalizeFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/casechanger/CapitalizeFormatterTest.java:14]:
```suggestion
    private CapitalizeFormatter formatter = new CapitalizeFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/UnitsToLatexFormatterTest.java [hunk 34]
```diff
@@ -10,12 +9,7 @@
  */
 public class UnitsToLatexFormatterTest {
 
-    private UnitsToLatexFormatter formatter;
-
-    @BeforeEach
-    public void setUp() {
-        formatter = new UnitsToLatexFormatter();
-    }
+    private static final UnitsToLatexFormatter formatter = new UnitsToLatexFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/UnitsToLatexFormatterTest.java:12]:
```suggestion
    private UnitsToLatexFormatter formatter = new UnitsToLatexFormatter();
```

File: src/test/java/org/jabref/logic/formatter/bibtexfields/UnicodeToLatexFormatterTest.java [hunk 35]
```diff
@@ -1,18 +1,12 @@
...
 
 class UnicodeToLatexFormatterTest {
 
-    private UnicodeToLatexFormatter formatter;
-
-    @BeforeEach
-    void setUp() {
-        formatter = new UnicodeToLatexFormatter();
-    }
+    private static final UnicodeToLatexFormatter formatter = new UnicodeToLatexFormatter();
```
Comment (koppor) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/UnicodeToLatexFormatterTest.java:9]:
```suggestion
    private UnicodeToLatexFormatter formatter = new UnicodeToLatexFormatter();
```

File: src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java (hunk 1 above)
Comment (Davfon) [reply src/test/java/org/jabref/logic/bibtex/comparator/CrossRefEntryComparatorTest.java:12]: Thanks for the feedback! I'll revert these changes then and use the setup method for those formatter tests.

**Comment by Davfon** [comment]
> Some small comments, then it should be good to go.

Thanks for the comments and suggested changes. I addressed them in the last commits.

File: src/test/java/org/jabref/logic/bst/BibtexCaseChangersTest.java [hunk 36]
```diff
@@ -1,134 +1,148 @@
...
-    private void assertCaseChangerAllUppers(final String string, final String string2) {
-        assertEquals(string, BibtexCaseChanger.changeCase(string2, FORMAT_MODE.ALL_UPPERS));
+    @Disabled
+    @Test
+    public void testTitleCaseAllUppers() {
+        /* the real test would look like as follows. Also from the comment of issue 176, order reversed as the "should be" comes first */
+        // assertCaseChangerTitleUppers("This is a Simple Example {TITLE}", "This is a simple example {TITLE}");
+        // assertCaseChangerTitleUppers("This {IS} Another Simple Example Tit{LE}", "This {IS} another simple example tit{LE}");
+        // assertCaseChangerTitleUppers("{What ABOUT thIS} one?", "{What ABOUT thIS} one?");
+        // assertCaseChangerTitleUppers("{And {thIS} might {a{lso}} be possible}", "{And {thIS} might {a{lso}} be possible}")
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bst/BibtexCaseChangersTest.java:146]: In case a test is `@Disabled`, the contents of the test can go enabled. Since this is probably the last comment on this PR, I'll let it go and we will fix for ourselves.

**Review by koppor — APPROVED** [review]

**Review by koppor — APPROVED** [review]
Now good to go. Note that the changes of #7697 to `NoBibTexFieldCheckerTest` was better as that were paramterized tests.
